File: Algo/Leetcode/461HammingDistance.cpp
// a^b is the difference of two value. O(1) space and time.
class Solution {
public:
    int hammingDistance(int x, int y) {
        int s = x^y;
        int count = 0;
        for(int i=0; i<32; i++){
            if(s&1<<i) count++;
        }
        return count;
    }
};

File: Algo/Leetcode/461HammingDistance.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        diff = 0
        for i in range(32):
            if (x & 1<<i) != (y & 1<<i):
                diff += 1

        return diff